kube-client/src/client/config_ext.rs

```diff
@@ -232,8 +232,12 @@ impl Config {
     fn exec_identity_pem(&self) -> Option<Vec<u8>> {
         match Auth::try_from(&self.auth_info) {
             Ok(Auth::Certificate(client_certificate_data, client_key_data)) => {
+                const NEW_LINE: u8 = b'\n';
+
                 let mut buffer = client_key_data.expose_secret().as_bytes().to_vec();
+                buffer.push(NEW_LINE);
                 buffer.extend_from_slice(client_certificate_data.as_bytes());
+                buffer.push(NEW_LINE);
```

Comment:
what happens here if `client_certificate_data` already containes newlines?

Reply:
nm, answered in pr body 👍

Reply:
actually, there's still the question if rustls deals with this case

Reply:
just tested it with `cargo run --example pod_watcher --no-default-features --features=rustls-tls,latest,runtime`

rustls works with none, single and double new line at the end.

Just to be on the safe side as we don't know if openssl would break in future if certs ends with double newline, we could just

```rust
if !buffer.ends_with(&[NEW_LINE]) {
  buffer.push(NEW_LINE);
}```

Reply:
if it's ok with both stack as it stands then everyone probably have do deal with newline padding here in the same way so i think it's ok to elide this check